hw/ip/otp_ctrl/data/otp_ctrl.hjson

```diff
@@ -1804,8 +1798,267 @@
       }
     ],
 
-    // these CSRs are defined in a separate hjson specific to the closed source OTP wrapper.
+    // OTP wrapper-specific registers
     prim: [
+      { name: "CSR0",
+        desc: ""
+        swaccess: "rw",
+        hwaccess: "hro",
+        hwext:    "false",
+        hwqe:     "false",
+        fields: [
```

Comment:
Not a problem, but I'm slightly surprised that we expose the field structure of aliased registers. Do we need that for technical reasons? Otherwise, I wonder whether it would be better to expose possibly different field structures. We could even say that the alias target is assumed to have a single 32-bit field?

Reply:
The motivation to do this is the following:

- Technical: this ensures the register structure is identical between open and closed source, which allows us to mix and match both open-source and closed-source DV code referencing the same registers and fields (both with open-source and closed-source names). Likewise, SW masks and access permissions are known on the open-source side for all registers.
- Project: especially the point about SW masks and known HW interface structure are important for OT as a project, since we have some software binaries that we have to be able to build off open-source code/headers only. 

We could loosen up the strictness a bit as you say, but then we would loose some of the guarantees that gives us...

Reply:
Side note: 

As for top-level specific variations of the hjson, I was thinking that this could be handled by passing IPs with such registers through `IPgen`. That would not be far from the current state since only `flash_ctrl` and `otp_ctrl` expose such registers at the moment - and the corresponding Hjsons are already templated. 

So the move to `IPgen` would have to be made anyways at some point in order to clean this up. `ast` is handled differently at the moment since it is not comportable - but since that is tailored towards a specific top-level, it already lives in a top-level specific location.

Reply:
Cool, this makes sense. Thanks for explaining what's going on.